move/move-ibc/sources/ibc.move

```diff
@@ -786,16 +775,15 @@ module IBC::Core {
             counterparty_version
         );
 
-        if (!verify_channel_state(
+        let err = verify_channel_state(
             &connection,
             proof_height,
             proof_init,
             *channel::counterparty_port_id(&counterparty),
             *channel::counterparty_channel_id(&counterparty),
             bcs::to_bytes(&expected_channel)
-        )) {
-            abort(E_INVALID_PROOF)
-        };
+        );
+        assert!(err == 0, E_INVALID_PROOF);
```

Comment:
why not abort with `err` here?

Reply:
errors are kinda garbage right now, proper error handling will be my next thing after voyager integration